test/runner.js

```diff
@@ -48,13 +50,41 @@ if ( process.env.CIRCLECI ) {
 mocha.suite.beforeAll( boot.before );
 mocha.suite.afterAll( boot.after );
 
-files = program.args;
-if ( files.length === 0 ) {
-	files = glob.sync( process.env.TEST_ROOT + '/**/test/*.@(js|jsx)' );
-	if ( program.nodeTotal > 1 ) {
-		files = files.filter( ( file, index ) => index % program.nodeTotal === program.nodeIndex );
+files = program.args.length ? program.args : [ process.env.TEST_ROOT ];
+files = files.reduce( ( memo, filePath ) => {
```

Comment:
We could refactor a bit this code now. It grew quite big recently. What do you think about moving that logic inside one of dependent file and using something similar to `flowRight` which would call functions like:

```js
executeUntil( validateTestRoot, appendIndividualFile, appendDiscoveredFiles );
```

Then we would have code chunks grouped by unique use cases and we don't need to add inline comments, because functions names are self-explanatory.